tests/unit/pluginhandler/test_patcher.py

```diff
@@ -117,40 +117,133 @@ def mock_partpatcher():
 class TestPrimeTypeExcludesPatching:
 
     scenarios = (
-        ("kernel", dict(snap_type="kernel", snap_name="test-snap")),
-        ("gadget", dict(snap_type="gadget", snap_name="test-snap")),
-        ("base", dict(snap_type="base", snap_name="core18")),
-        ("os", dict(snap_type="os", snap_name="test-snap")),
+        (
+            "kernel",
+            dict(
+                snap_type="kernel",
+                snap_name="test-snap",
+                confinement=None,
+                build_attributes=None,
+            ),
+        ),
+        (
+            "gadget",
+            dict(
+                snap_type="gadget",
+                snap_name="test-snap",
+                confinement=None,
+                build_attributes=None,
+            ),
+        ),
+        (
+            "base",
+            dict(
+                snap_type="base",
+                snap_name="core18",
+                confinement=None,
+                build_attributes=None,
+            ),
+        ),
+        (
+            "os",
+            dict(
+                snap_type="os",
+                snap_name="test-snap",
+                confinement=None,
```

Comment:
`confinement=strict` may avoid some issues at some point

Reply:
That feels a bit... improper :P . But sure, fixed.